opencti-platform/opencti-graphql/src/domain/settings.js

```diff
@@ -106,3 +110,63 @@ export const settingsEditField = async (context, user, settingsId, input) => {
   const updatedSettings = await getSettings(context);
   return notify(BUS_TOPICS.Settings.EDIT_TOPIC, updatedSettings, user);
 };
+
+export const getMessages = (settings) => {
+  return JSON.parse(settings.messages ?? '[]');
+};
+
+export const settingEditMessage = async (context, user, settingsId, message) => {
+  const messageToStore = {
+    ...message,
+    updated_at: now()
+  };
+  const settings = await getEntityFromCache(context, user, ENTITY_TYPE_SETTINGS);
+  const messages = getMessages(settings);
+
+  const existingIdx = messages.findIndex((m) => m.id === message.id);
+  if (existingIdx > -1) {
+    messages[existingIdx] = messageToStore;
+  } else {
+    messages.push({
+      ...messageToStore,
+      id: generateInternalId()
+    });
+  }
+  const patch = { messages: JSON.stringify(messages) };
+  const { element } = await patchAttribute(context, user, settingsId, ENTITY_TYPE_SETTINGS, patch);
+  return notify(BUS_TOPICS[ENTITY_TYPE_SETTINGS].EDIT_TOPIC, element, user);
+};
+
+export const settingDeleteMessage = async (context, user, settingsId, messageId) => {
+  const settings = await getEntityFromCache(context, user, ENTITY_TYPE_SETTINGS);
+  const messages = getMessages(settings);
+
+  const existingIdx = messages.findIndex((m) => m.id === messageId);
+  if (existingIdx > -1) {
+    messages.splice(existingIdx, 1);
+  } else {
+    throw UnsupportedError('This message does not exist', { messageId });
+  }
+  const patch = { messages: JSON.stringify(messages) };
+  const { element } = await patchAttribute(context, user, settingsId, ENTITY_TYPE_SETTINGS, patch);
+  return notify(BUS_TOPICS[ENTITY_TYPE_SETTINGS].EDIT_TOPIC, element, user);
+};
+
+// -- AJV --
+
+export const settingsMessages = {
+  type: 'array',
+  items: {
+    type: 'object',
+    properties: {
+      id: {
+        type: 'string',
+        minLength: 1
+      },
+      message: { type: 'string' },
+      activated: { type: 'boolean' },
+      updated_at: { type: 'string' },
```

Comment:
shouldn't we have `dismissible` here ? It doesn't throw an error when we set a property not defined in the schema ?

Reply:
My bad, I did not secure the model in the right way